src/app/sources/DeleteSourceForm.tsx

```diff
@@ -0,0 +1,39 @@
+"use client";
+
+import { deleteSourceAction } from "@/lib/actions/sources";
+import { useActionState } from "react";
+
+export default function DeleteSourceForm({
+  sourceId,
+}: {
+  sourceId: string | number;
+}) {
+  const [state, formAction, pending] = useActionState(
+    deleteSourceAction,
+    undefined,
+  );
+
+  return (
+    <form action={formAction} className="space-y-2">
+      <input type="hidden" name="sourceId" value={sourceId} />
+
+      <button
+        type="submit"
+        className="text-red-500 hover:text-red-600 cursor-pointer"
+        disabled={pending}
+      >
+        Delete
+      </button>
+
+      {state?.properties?.sourceId?.errors && (
+        <ul className="list-disc list-inside text-red-500">
+          {state.properties.sourceId.errors.map((error) => (
+            <li key={error} className="text-red-500">
+              {error}
+            </li>
+          ))}
+        </ul>
+      )}
```

Comment:
Admittedly this is stupid, but fine for now. You wouldn't get a validation error on the presence of a hidden input. This stupidity effects all hidden input removal forms, below.

Tracking #11